src/app/portlets/dot-apps/dot-apps-export-dialog/dot-apps-export-dialog.component.ts

```diff
@@ -0,0 +1,116 @@
+import { Component, OnInit, OnDestroy, Input } from '@angular/core';
+import { FormGroup, FormBuilder, FormControl, Validators } from '@angular/forms';
+import { DotDialogActions } from '@components/dot-dialog/dot-dialog.component';
+import { DotAppsService } from '@services/dot-apps/dot-apps.service';
+import { DotMessageService } from '@services/dot-message/dot-messages.service';
+import {
+    DotApps,
+    DotAppsExportConfiguration,
+    DotAppsSites
+} from '@shared/models/dot-apps/dot-apps.model';
+import { Subject } from 'rxjs/internal/Subject';
+import { takeUntil } from 'rxjs/operators';
+
+@Component({
+    selector: 'dot-apps-export-dialog',
+    templateUrl: './dot-apps-export-dialog.component.html'
+})
+export class DotAppsExportDialogComponent implements OnInit, OnDestroy {
+    @Input() app?: DotApps;
+    @Input() site?: DotAppsSites;
+
+    showExportDialog = false;
+    form: FormGroup;
+    dialogExportActions: DotDialogActions;
+    exportErrorMessage: string;
+
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private dotAppsService: DotAppsService,
+        private dotMessageService: DotMessageService,
+        private fb: FormBuilder
+    ) {}
+
+    ngOnInit() {
+        this.form = this.fb.group({
+            password: new FormControl('', Validators.required)
+        });
+
+        this.form.valueChanges.pipe(takeUntil(this.destroy$)).subscribe(() => {
+            this.dialogExportActions = {
+                ...this.dialogExportActions,
+                accept: {
+                    ...this.dialogExportActions.accept,
+                    disabled: !this.form.valid
+                }
+            };
+        });
+        this.setExportDialogActions();
+    }
+
+    ngOnDestroy(): void {
+        this.destroy$.next(true);
+        this.destroy$.complete();
+    }
+
+    /**
+     * Close the dialog and clear the form
+     *
+     * @memberof DotAppsConfigurationComponent
+     */
+    closeExportDialog(): void {
+        this.exportErrorMessage = '';
+        this.form.reset();
+        this.site = null;
+        this.showExportDialog = false;
+    }
+
+    private setExportDialogActions(): void {
+        this.dialogExportActions = {
+            accept: {
+                action: () => {
+                    const requestConfiguration: DotAppsExportConfiguration = {
+                        password: this.form.value.password,
+                        exportAll: this.app ? false : true,
+                        appKeysBySite: this.site
+                            ? { [this.site.id]: [this.app.key] }
```

Comment:
If we only using the side id, shouldn't we just pass the id?

Reply:
no, that endpoints supports multiple params to get or not a single site from a configuration app or all of them if not specified

https://authoring.dotcms.com/devwiki/apps-rest-api